src/Microsoft.AspNet.Razor/Chunks/ChunkTreeBuilder.cs

```diff
@@ -69,22 +70,39 @@ public void AddRemoveTagHelperChunk(string lookupText, SyntaxTreeNode associatio
 
         public void AddLiteralChunk(string literal, SyntaxTreeNode association)
         {
-            // If the previous chunk was also a LiteralChunk, append the content of the current node to the previous one.
-            var literalChunk = _lastChunk as LiteralChunk;
-            if (literalChunk != null)
+            ParentLiteralChunk parentLiteralChunk;
```

Comment:
I don't like it better. :-1: I wrote the code the way I did to make it clear that there are three distinct states, and make it easy to grasp what to do in each case. I added some comments here that hopefully makes it even better.